src/main/java/edu/byu/cs/controller/SubmissionController.java

```diff
@@ -90,7 +91,8 @@ private static void startGrader(String netId, Phase phase, String repoUrl) {
     }
 
     private static void updateRepoFromCanvas(User user, Request req) throws CanvasException {
-        String newRepoUrl = CanvasIntegration.getGitRepo(user.canvasUserId());
+        CanvasIntegration canvas = new CanvasIntegrationImpl();
```

Comment:
Michael has a getter in `CanvasIntegration`, `getCanvasIntegration()`. If we use that instead of specifying `new CanvasIntegrationImpl()` it will make this code more compatible with mocking

Reply:
done and done